sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSqlTable.java

```diff
@@ -36,4 +37,9 @@
 
   /** Get the schema info of the table. */
   Schema getSchema();
+
+  /** Estimates the number of rows or returns null if there is no estimation. */
+  default Double getRowCount(PipelineOptions options) {
```

Comment:
Would it make sense to create our own version of `Statistic` for extensibility (only with row count for now)?

Reply:
Also why Double?

Reply:
The reason for double is because of calcite Statistic. It returns Double for row count.

Reply:
I suggest we use `long` or something else that is exact for now. It will be straightforward to go to approximate types if we need to, but not other way around